src/jdk.compiler/share/classes/com/sun/tools/javac/launcher/MemoryClassLoader.java

```diff
@@ -271,6 +267,25 @@ public URL nextElement() {
         };
     }
 
+    /**
+     * Resolves a "resource name" (as used in the getResource* methods)
+     * to an existing file relative to source root path, or null otherwise.
+     *
+     * @param name the resource name
+     * @return the URL of the resource, or null
+     */
+    private URL toResourceInRootPath(String name) {
```

Comment:
Is this method expected to return `URL` to a `.class` file too? The reason I ask this is, I notice that for classes that belong within the source java file we use a different protocol scheme for the URIs.

Reply:
Every file that is resolvable under the computed source root path is expected to be returned by this method. This includes `.class` files. But you maybe right that such files should be excluded; or at least be handled the application class loader. Have to write a few more tests to give a better answer.